File: app/src/main/java/org/cerion/stockcharts/common/BindingUtils.java
package org.cerion.stockcharts.common;

import androidx.databinding.BindingAdapter;
import androidx.databinding.InverseBindingAdapter;
import androidx.databinding.InverseBindingListener;
import android.util.Log;
import android.widget.Spinner;

public class BindingUtils {
    private static final String TAG = BindingUtils.class.getSimpleName();

    @BindingAdapter({"selection"})
    public static void setSelection(final Spinner spinner, final int position) {
        Log.d(TAG, "setSelection");
        spinner.post(new Runnable() {
            @Override
            public void run() {
                spinner.setSelection(position);
            }
        });
    }

    /*


    @InverseBindingAdapter(attribute = "selection")
    public static int getSelection(Spinner spinner) {
        Log.d(TAG, "bindSelectionInverseAdapter");
        return spinner.getSelectedItemPosition();
    }

    @BindingAdapter(value = "selectionAttrChanged")
    public static void bindSelectionChanged(Spinner spinnner, final InverseBindingListener newAttrChanged) {
        Log.d(TAG, "bindSelectionChanged");

        AdapterView.OnItemSelectedListener listener = new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                newAttrChanged.onChange();
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {
                newAttrChanged.onChange();
            }
        };

        spinnner.setOnItemSelectedListener(listener);
    }
    */
}

File: app/src/main/java/org/cerion/stockcharts/common/BindingUtils.kt
package org.cerion.stockcharts.common

import android.widget.CheckBox
import androidx.lifecycle.LifecycleOwner
import androidx.lifecycle.MutableLiveData

object BindingUtils {

    fun setTwoWayBinding(lifeCycleOwner: LifecycleOwner, checkBox: CheckBox, data: MutableLiveData<Boolean>) {
        data.observe(lifeCycleOwner) {
            checkBox.isChecked = it
        }

        checkBox.setOnClickListener {
            val checkbox = it as CheckBox
            data.value = checkbox.isChecked
        }
    }
}